src/renderer/src/routes/modals/SellTokensDeposit/SellTokensDepositModal.tsx

```diff
@@ -0,0 +1,465 @@
+      account,
+      isFeeLoading: false,
+    }
+  )
+
+  const { data: balanceData, isLoading: isBalanceLoading } = useBalance(actionData.account)
+
+  const service = useMemo(
+    () => (actionData?.account ? bsAggregator.blockchainServicesByName[actionData.account.blockchain] : undefined),
+    [actionData.account]
+  )
+
+  const hasCalculableFee = service ? isCalculableFee(service) : false
+  const isRecipientDisabled = !actionData.account
+  const isInvalidForm =
+    isRecipientDisabled || !actionData.address || !actionData.amount || !actionData.token || !service
+
+  const handleClose = () => {
+    modalNavigate(-1)
+    dispatch(buyAndSellTokensReducerActions.saveDepositActionsData(actionData))
+  }
+
+  const getTransfersData = async () => {
+    const { account, address } = actionData
+    const token = actionData.token?.token
+    const encryptedPassword = currentLoginSessionRef.current?.encryptedPassword
+
+    if (!encryptedPassword || isInvalidForm) return
+
+    const intents: IntentTransferParam[] = [
+      {
+        amount: actionData.amount,
+        receiverAddress: address,
+        tokenHash: token!.hash,
+        tokenDecimals: token!.decimals,
+      },
+    ]
+
+    const key = await window.api.sendAsync('decryptBasedEncryptedSecret', {
+      value: account!.encryptedKey!,
+      encryptedSecret: encryptedPassword,
+    })
+
+    let serviceAccount: Account<TBlockchainServiceKey>
+
+    if (account!.type === 'hardware' && hasLedger(service)) {
+      serviceAccount = service.generateAccountFromPublicKey(key)
+      serviceAccount.isHardware = true
+      serviceAccount.bip44Path = AccountHelper.getBip44Path(service, account!.order)
+    } else {
+      serviceAccount = service.generateAccountFromKey(key)
+    }
+
+    return {
+      serviceAccount,
+      intents,
+    }
+  }
+
+  const handleChangeAccount = (account: IAccountState) => {
+    setData({ account })
+  }
+
+  const handleChangeAmount = (value: string) => {
+    setData({ amount: NumberHelper.formatString(value, actionData.token?.token?.decimals, 24) })
+  }
+
+  const handleChangeAddress = (event: ChangeEvent<HTMLInputElement>) => {
+    const address = UtilsHelper.removeSpecialCharacters(event.target.value, { allowSpaces: false })
+
+    setData({ address })
+  }
+
+  const handleChangeToken = (token: Token) => {
+    const tokenHash = UtilsHelper.normalizeHash(token.hash)
+    const tokenBalance = balanceData?.tokensBalances.find(
+      tokenBalance => UtilsHelper.normalizeHash(tokenBalance.token.hash) === tokenHash
+    )
+
+    setData({ token: tokenBalance, amount: '' })
+  }
+
+  const handleSubmit = async () => {
+    const transfersData = await getTransfersData()
+    const { address, amount, token, account } = actionData
+
+    if (!transfersData || actionData.isFeeLoading || isInvalidForm) return
+
+    try {
+      const { serviceAccount, intents } = transfersData
+      const transactionHashes = await service.transfer({
```

Comment:
Can't you get the first item since you know that it will always be one transfer ?
```suggestion
      const [ transactionHash ] = await service.transfer({
```